microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/MicroProfileHover.java

```diff
@@ -191,16 +192,22 @@ private static Hover getPropertyValueExpressionHover(Node node, MicroProfileProj
 		if (referencedProp == null) {
 			return null;
 		}
+
+		PropertyGraph graph = PropertyGraph.buildPropertyGraph(node.getOwnerModel());
+		
+		if (graph.isAcyclic()) {
+			String resolvedValue = propValExpr.getResolvedValue(graph, projectInfo);
+			if (resolvedValue != null) {
+				return getContentAsPlaintextHover(resolvedValue, node);
+			}
+		}
+
 		// Check property file for the value
 		for (Node modelChild: node.getOwnerModel().getChildren()) {
 			if (modelChild.getNodeType() == NodeType.PROPERTY) {
 				Property otherProperty = (Property) modelChild;
 				if (referencedProp.equals(otherProperty.getPropertyNameWithProfile())) {
-					Hover hover = new Hover();
-					// NOTE: not resolving property expressions recursively
-					hover.setContents(Either.forLeft(Collections.singletonList(Either.forLeft(otherProperty.getPropertyValue()))));
-					hover.setRange(PositionUtils.createRange(node));
-					return hover;
+					return getContentAsPlaintextHover(otherProperty.getPropertyValue(), node);
```

Comment:
It might make sense to return the hover as Markdown, and surround the value with backticks

Reply:
Relevant bug in implementation:

![Screenshot from 2020-07-31 11-33-57](https://user-images.githubusercontent.com/22376627/89051346-1e5d4800-d322-11ea-85cd-a3a888403aaa.png)

Reply:
I tried formatting the hover value with backticks in markdown, and it looked odd, so I kept it with no markdown. I fixed the multi line value issue and added a unit test for this.